### PR TITLE
Fixes for ONNXRuntime for compilation with gcc 14.2

### PR DESCRIPTION
**Pull request by davidrohr**

### CODE REVIEW

File: onnxruntime.sh [hunk 1]
```diff
@@ -35,8 +35,8 @@ cmake "$SOURCEDIR/cmake"
       ${PROTOBUF_ROOT:+-DProtobuf_PROTOC_EXECUTABLE=$PROTOBUF_ROOT/bin/protoc}        \
       ${RE2_ROOT:+-DRE2_INCLUDE_DIR=${RE2_ROOT}/include}                              \
       ${BOOST_ROOT:+-DBOOST_INCLUDE_DIR=${BOOST_ROOT}/include}                        \
-      -DCMAKE_CXX_FLAGS="$CXXFLAGS -Wno-unknown-warning -Wno-unknown-warning-option -Wno-error=unused-but-set-variable" \
-      -DCMAKE_C_FLAGS="$CFLAGS -Wno-unknown-warning -Wno-unknown-warning-option -Wno-error=unused-but-set-variable"
+      -DCMAKE_CXX_FLAGS="$CXXFLAGS -Wno-unknown-warning -Wno-unknown-warning-option -Wno-error=unused-but-set-variable -Wno-error=maybe-uninitialized -Wno-error=template-id-cdtor" \
```
Comment (aalkin) [inline onnxruntime.sh:38]: Shouldn't it be `-Wno-template-id-cdtor`?

**Comment by davidrohr** [comment]
Hm, but for sure what I have here works and is needed for gcc 14

**Review by ktf — COMMENTED** [review]
Seems not to work.

**Comment by davidrohr** [comment]
So, they added that option only with gcc 14, beforehand it is not available...
But then with gcc 14, this will be needed...
Marking this as draft for now until we bump to gcc 14.

**Comment by davidrohr** [comment]
closing this, will open PR for GCC 14.2 which will include this

**Pull request closed without merging**